mne/viz/epochs.py

```diff
@@ -536,3 +724,172 @@ def plot_epochs_psd(epochs, fmin=0, fmax=np.inf, proj=False, n_fft=256,
+                                params['n_epochs'])
+
+            lines[ii].set_segments(segments)
+            vars(lines[ii])['ch_name'] = ch_name
+            color = params['color'][types[ch_ind]]
+            vars(lines[ii])['def_color'] = params['color'][types[ch_ind]]
+            lines[ii].set_color(color)
+        else:
+            # "remove" lines
+            lines[ii].set_xdata([])
+            lines[ii].set_ydata([])
+    # deal with event lines
+    if params['epoch_times'] is not None:
+        # find events in the time window
+        epoch_times = params['epoch_times']
+        event_line = params['event_line']
+        xs = list()
+        ys = list()
+        for tt in epoch_times:
+            xs += [tt, tt, np.nan]
+            ylim = params['ax'].get_ylim()
+            ys += [ylim[0], ylim[1], np.nan]
+        event_line.set_xdata(xs)
+        event_line.set_ydata(ys)
+
+    # finalize plot
+    params['ax'].set_xlim(params['times'][0],
+                          params['times'][0] + params['duration'], False)
+    params['ax'].set_yticklabels(tick_list)
+    params['vsel_patch'].set_y(params['ch_start'])
+    params['fig'].canvas.draw()
+
+
+def _plot_onscroll(event, params):
+    """
+    Scroll events.
+    """
+    orig_start = params['ch_start']
+    if event.step < 0:
+        params['ch_start'] = min(params['ch_start'] + params['n_channels'],
+                                 len(params['ch_names']) -
+                                 params['n_channels'])
+    else:  # event.key == 'up':
+        params['ch_start'] = max(params['ch_start'] - params['n_channels'], 0)
+    if orig_start != params['ch_start']:
+        _channels_changed(params)
+
+
+def _plot_window(value, params):
+    """
+    Deal with horizontal epoch window.
+    """
+    max_times = len(params['times']) - params['duration']
+    if value > max_times:
+        value = len(params['times']) - params['duration']
+    if value < 0:
+        value = 0
+    if params['t_start'] != value:
+        params['t_start'] = value
+        params['hsel_patch'].set_x(value)
+        _plot_traces(params)
+
+
+def _mouse_click(event, params):
+    """
+    Mouse click events.
+    """
+    if event.inaxes is None or event.button != 1:
+        return
+    # vertical scroll bar changed
+    if event.inaxes == params['ax_vscroll']:
+        ch_start = max(int(event.ydata) - params['n_channels'] // 2, 0)
+        if params['ch_start'] != ch_start:
+            params['ch_start'] = ch_start
+            _plot_traces(params)
+    # horizontal scroll bar changed
+    elif event.inaxes == params['ax_hscroll']:
+        # find the closest epoch time
+        times = params['epoch_times']
+        xdata = times.flat[np.abs(times - event.xdata).argmin()]
+        _plot_window(xdata, params)
+    # main axes
+    elif event.inaxes == params['ax']:
+        _pick_bad_epochs(event, params)
+
+
+def _plot_onkey(event, params):
+    """
+    Key presses.
+    """
```

Comment:
But it in one line and say, function to handle key presses. One grammatically complete sentence, even if short would be desriable for a 1line doc.